builtin/providers/spotinst/resource_spotinst_aws_group.go

```diff
@@ -0,0 +1,2077 @@
+			"ondemand_count":             g.Strategy.OnDemandCount,
+			"availability_vs_cost":       g.Strategy.AvailabilityVsCost,
+			"draining_timeout":           g.Strategy.DrainingTimeout,
+			"utilize_reserved_instances": g.Strategy.UtilizeReservedInstances,
+			"fallback_to_ondemand":       g.Strategy.FallbackToOnDemand,
+		})
+		d.Set("strategy", strategy)
+
+		// Set the launch specification.
+		lspec := make([]map[string]interface{}, 0, 1)
+		l := map[string]interface{}{
+			"health_check_grace_period": g.Compute.LaunchSpecification.HealthCheckGracePeriod,
+			"health_check_type":         g.Compute.LaunchSpecification.HealthCheckType,
+			"image_id":                  g.Compute.LaunchSpecification.ImageID,
+			"key_pair":                  g.Compute.LaunchSpecification.KeyPair,
+			"load_balancer_names":       g.Compute.LaunchSpecification.LoadBalancerNames,
+			"monitoring":                g.Compute.LaunchSpecification.Monitoring,
+			"security_group_ids":        g.Compute.LaunchSpecification.SecurityGroupIDs,
+			"user_data":                 g.Compute.LaunchSpecification.UserData,
+		}
+		if g.Compute.LaunchSpecification.IamInstanceProfile != nil {
+			if g.Compute.LaunchSpecification.IamInstanceProfile.Arn != nil {
+				l["iam_instance_profile"] = g.Compute.LaunchSpecification.IamInstanceProfile.Arn
+			} else {
+				l["iam_instance_profile"] = g.Compute.LaunchSpecification.IamInstanceProfile.Name
+			}
+		}
+		lspec = append(lspec, l)
+		d.Set("launch_specification", lspec)
+
+		// Set the load balancers.
+		if g.Compute.LaunchSpecification.LoadBalancersConfig != nil {
+			lbs := make([]map[string]interface{}, 0, len(g.Compute.LaunchSpecification.LoadBalancersConfig.LoadBalancers))
+			for _, lb := range g.Compute.LaunchSpecification.LoadBalancersConfig.LoadBalancers {
+				lbs = append(lbs, map[string]interface{}{
+					"name": lb.Name,
+					"arn":  lb.Arn,
+					"type": lb.Type,
+				})
+			}
+			d.Set("load_balancer", lbs)
+		}
+
+		// Set the availability zones.
+		zones := make([]map[string]interface{}, 0, len(g.Compute.AvailabilityZones))
+		for _, z := range g.Compute.AvailabilityZones {
+			zones = append(zones, map[string]interface{}{
+				"name":      z.Name,
+				"subnet_id": z.SubnetID,
+			})
+		}
+		d.Set("availability_zone", zones)
+
+		// Set the EBS volume pool.
+		volumes := make([]map[string]interface{}, 0, len(g.Compute.EBSVolumePool))
+		for _, v := range g.Compute.EBSVolumePool {
+			volumes = append(volumes, map[string]interface{}{
+				"device_name": v.DeviceName,
+				"volume_ids":  v.VolumeIDs,
+			})
+		}
+		d.Set("hot_ebs_volume", volumes)
+
+		// Set the signals.
+		signals := make([]map[string]interface{}, 0, len(g.Strategy.Signals))
+		for _, s := range g.Strategy.Signals {
+			signals = append(signals, map[string]interface{}{
+				"name": s.Name,
+			})
+		}
+		d.Set("signal", signals)
+
+		// Set the scheduled tasks.
+		tasks := make([]map[string]interface{}, 0, len(g.Scheduling.Tasks))
```

Comment:
would love to see this split out into a [flatten style func](https://github.com/hashicorp/terraform/blob/master/builtin/providers/azurerm/resource_arm_virtual_machine.go#L601)